client/proxy/proxy.go

```diff
@@ -34,76 +33,23 @@ type Client interface {
 
 // NewManagerClient creates a new ManagerClient instance. It trims the apiVersion of leading and trailing slashes
 // and the base path of trailing slashes to ensure consistency
-func NewManagerClient(base url.URL, apiVersion string, client *http.Client) *ManagerClient {
-	base.Path = strings.TrimSuffix(base.Path, "/")
-	return &ManagerClient{
-		base:             base,
-		versionedAPIPath: strings.TrimPrefix(strings.TrimSuffix(apiVersion, "/"), "/"),
-		client:           client,
-	}
-}
-
-func (m *ManagerClient) toCurl(request *http.Request, body string) string {
-
```

Comment:
> From https://github.com/istio/manager/issues/645 -  istio.io/manager/client/proxy/ManagerClient should log HTTP, similar to what the Kubernetes clients do.

This change uses the Kubernetes client so the logging behavior is _exactly_ like what other Kubernetes clients do, e.g.

```bash
$ istioctl -v=999 create -f route-rule-delay.yaml
I0505 09:35:37.315569  132334 loader.go:354] Config loaded from file /usr/local/google/home/jasonyoung/.kube/config
I0505 09:35:37.316597  132334 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: istioctl/v0.0.0 (linux/amd64) kubernetes/$Format" -H "Authorization: Basic YWRtaW46WXdRcEVvRzVLV2tzZ1hueQ==" https://35.184.13.238/apis/extensions/v1beta1/thirdpartyresources/istio-config.istio.io
I0505 09:35:37.544263  132334 round_trippers.go:417] GET https://35.184.13.238/apis/extensions/v1beta1/thirdpartyresources/istio-config.istio.io 200 OK in 227 milliseconds
I0505 09:35:37.544285  132334 round_trippers.go:423] Response Headers:
I0505 09:35:37.544298  132334 round_trippers.go:426]     Content-Type: application/json
I0505 09:35:37.544306  132334 round_trippers.go:426]     Content-Length: 369
I0505 09:35:37.544315  132334 round_trippers.go:426]     Date: Fri, 05 May 2017 16:35:37 GMT
I0505 09:35:37.544345  132334 request.go:991] Response Body: {"kind":"ThirdPartyResource","apiVersion":"extensions/v1beta1","metadata":{"name":"istio-config.istio.io","selfLink":"/apis/extensions/v1beta1/thirdpartyresourcesistio-config.istio.io","uid":"21f2d9e2-2f6c-11e7-a278-42010a80017c","resourceVersion":"1471","creationTimestamp":"2017-05-02T19:18:33Z"},"description":"Istio configuration","versions":[{"name":"v1alpha1"}]}
I0505 09:35:37.544539  132334 client.go:173] Resource already exists: "istio-config.istio.io"
I0505 09:35:37.544557  132334 client.go:197] Checking for TPR resources
I0505 09:35:37.544607  132334 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json, */*" -H "Authorization: Basic YWRtaW46WXdRcEVvRzVLV2tzZ1hueQ==" https://35.184.13.238/apis/istio.io/v1alpha1/istioconfigs
I0505 09:35:37.584012  132334 round_trippers.go:417] GET https://35.184.13.238/apis/istio.io/v1alpha1/istioconfigs 200 OK in 39 milliseconds
I0505 09:35:37.584032  132334 round_trippers.go:423] Response Headers:
I0505 09:35:37.584039  132334 round_trippers.go:426]     Content-Length: 2013
I0505 09:35:37.584044  132334 round_trippers.go:426]     Date: Fri, 05 May 2017 16:35:37 GMT
I0505 09:35:37.584049  132334 round_trippers.go:426]     Content-Type: application/json
I0505 09:35:37.584115  132334 request.go:991] Response Body: {"kind":"IstioConfigList","items":[{"apiVersion":"istio.io/v1alpha1","kind":"IstioConfig","metadata":{"name":"route-rule-details-default","namespace":"default","selfLink":"/apis/istio.io/v1alpha1/namespaces/default/istioconfigs/route-rule-details-default","uid":"d80cdbdd-3136-11e7-a278-42010a80017c","resourceVersion":"424970","creationTimestamp":"2017-05-05T02:02:08Z"},"spec":{"destination":"details.default.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v1"}}]}},{"apiVersion":"istio.io/v1alpha1","kind":"IstioConfig","metadata":{"name":"route-rule-productpage-default","namespace":"default","selfLink":"/apis/istio.io/v1alpha1/namespaces/default/istioconfigs/route-rule-productpage-default","uid":"d7f7929b-3136-11e7-a278-42010a80017c","resourceVersion":"424967","creationTimestamp":"2017-05-05T02:02:07Z"},"spec":{"destination":"productpage.default.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v1"}}]}},{"apiVersion":"istio.io/v1alpha1","kind":"IstioConfig","metadata":{"name":"route-rule-ratings-default","namespace":"default","selfLink":"/apis/istio.io/v1alpha1/namespaces/default/istioconfigs/route-rule-ratings-default","uid":"d805db3d-3136-11e7-a278-42010a80017c","resourceVersion":"424969","creationTimestamp":"2017-05-05T02:02:08Z"},"spec":{"destination":"ratings.default.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v1"}}]}},{"apiVersion":"istio.io/v1alpha1","kind":"IstioConfig","metadata":{"name":"route-rule-reviews-default","namespace":"default","selfLink":"/apis/istio.io/v1alpha1/namespaces/default/istioconfigs/route-rule-reviews-default","uid":"d43208f5-313a-11e7-a278-42010a80017c","resourceVersion":"427446","creationTimestamp":"2017-05-05T02:30:39Z"},"spec":{"destination":"reviews.default.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v2"},"weight":50},{"tags":{"version":"v3"},"weight":50}]}}],"metadata":{"selfLink":"/apis/istio.io/v1alpha1/istioconfigs","resourceVersion":"501380"},"apiVersion":"istio.io/v1alpha1"}
I0505 09:35:37.584653  132334 decoder.go:224] decoding stream as YAML
I0505 09:35:37.585543  132334 request.go:991] Request Body: {"type":"route-rule","name":"ratings-test-delay","spec":{"destination":"ratings.default.svc.cluster.local","httpFault":{"delay":{"fixedDelay":"7s","percent":100}},"match":{"httpHeaders":{"Cookie":{"regex":"^(.*?;)?(user=test-user)(;.*)?$"}}},"precedence":2,"route":[{"tags":{"version":"v1"}}]}}
I0505 09:35:37.585614  132334 round_trippers.go:398] curl -k -v -XPOST  -H "Accept: application/json, */*" -H "Content-Type: application/json" -H "Authorization: Basic YWRtaW46WXdRcEVvRzVLV2tzZ1hueQ==" https://35.184.13.238/api/v1/namespaces/default/services/istio-manager:8081/proxy/v1alpha1/config/route-rule/default/ratings-test-delay
I0505 09:35:37.635058  132334 round_trippers.go:417] POST https://35.184.13.238/api/v1/namespaces/default/services/istio-manager:8081/proxy/v1alpha1/config/route-rule/default/ratings-test-delay 201 Created in 49 milliseconds
I0505 09:35:37.635480  132334 round_trippers.go:423] Response Headers:
I0505 09:35:37.635807  132334 round_trippers.go:426]     Content-Type: application/json
I0505 09:35:37.636078  132334 round_trippers.go:426]     Date: Fri, 05 May 2017 16:35:37 GMT
I0505 09:35:37.636509  132334 round_trippers.go:426]     Content-Length: 436
I0505 09:35:37.636834  132334 request.go:991] Response Body: {
  "type": "route-rule",
  "name": "ratings-test-delay",
  "spec": {
   "destination": "ratings.default.svc.cluster.local",
   "httpFault": {
    "delay": {
     "fixedDelay": "7s",
     "percent": 100
    }
   },
   "match": {
    "httpHeaders": {
     "Cookie": {
      "regex": "^(.*?;)?(user=test-user)(;.*)?$"
     }
    }
   },
   "precedence": 2,
   "route": [
    {
     "tags": {
      "version": "v1"
     }
    }
   ]
  }
 }
Created config: route-rule ratings-test-delay
```